docker/Dockerfile

```diff
@@ -0,0 +1,94 @@
+FROM ubuntu:latest@sha256:3f119dc0737f57f704ebecac8a6d8477b0f6ca1ca0332c7ee1395ed2c6a82be7
+
+LABEL author="Vincent Voyer <vincent@zeroload.net>"
+LABEL maintainers="Serban Ghita <serbanghita@gmail.com>, David Catalan <catalan.david@gmail.com>"
+
+# Env variables
+ENV LC_ALL=C
+ENV DISPLAY=:99
+ENV SCREEN_GEOMETRY=1024x768x16
+
+
+# Expose Selenium web console port
+EXPOSE 4444
+
+
+# Update packages
+RUN apt-get -qqy update
+
+
+# Install commons
+RUN apt-get -qqy install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    software-properties-common \
+    sudo \
+    openjdk-8-jre-headless \
+    wget \
+    xvfb \
+    xfonts-100dpi \
+    xfonts-75dpi \
+    xfonts-scalable \
+    xfonts-cyrillic
+
+
+# Install Browser from vendor source
+# Strategies copied from official Selenium Docker images (https://github.com/SeleniumHQ/docker-selenium)
+
+## Install latest Firefox
+ARG FIREFOX_VERSION=latest
+RUN FIREFOX_DOWNLOAD_URL=$(if [ $FIREFOX_VERSION = "latest" ] || [ $FIREFOX_VERSION = "nightly-latest" ] || [ $FIREFOX_VERSION = "devedition-latest" ]; then echo "https://download.mozilla.org/?product=firefox-$FIREFOX_VERSION-ssl&os=linux64&lang=en-US"; else echo "https://download-installer.cdn.mozilla.net/pub/firefox/releases/$FIREFOX_VERSION/linux-x86_64/en-US/firefox-$FIREFOX_VERSION.tar.bz2"; fi) \
+  && apt-get update -qqy \
+  && apt-get -qqy --no-install-recommends install firefox \
+  && rm -rf /var/lib/apt/lists/* /var/cache/apt/* \
+  && wget --no-verbose -O /tmp/firefox.tar.bz2 $FIREFOX_DOWNLOAD_URL \
+  && apt-get -y purge firefox \
+  && rm -rf /opt/firefox \
+  && tar -C /opt -xjf /tmp/firefox.tar.bz2 \
+  && rm /tmp/firefox.tar.bz2 \
+  && mv /opt/firefox /opt/firefox-$FIREFOX_VERSION \
+  && ln -fs /opt/firefox-$FIREFOX_VERSION/firefox /usr/bin/firefox
+
+## Install latest Google Chrome
+ARG CHROME_VERSION="google-chrome-stable"
+RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
+  && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list \
+  && apt-get update -qqy \
+  && apt-get -qqy install \
+    ${CHROME_VERSION:-google-chrome-stable} \
+  && rm /etc/apt/sources.list.d/google-chrome.list \
+  && rm -rf /var/lib/apt/lists/* /var/cache/apt/*
+
+## Chrome Launch Script Wrapper (forces no-sandbox so no need to use --privileged parameter)
+COPY ./scripts/wrap_chrome_binary /opt/bin/wrap_chrome_binary
```

Comment:
`Note: --no-sandbox is not needed if you properly setup a user in the container.` [0]
Previously I was adding user `node`

[0] https://developers.google.com/web/updates/2017/04/headless-chrome

Reply:
I am also adding and using a non-root user (otherwise Chrome would not start). Though, no-sandbox is required (or docker privileged parameter).

Reply:
@catalan-adobe To give you some context, the reason why tests previously worked with the current `Dockerfile` is that I ran them in headless mode for both browsers. If you clone `serbanghita/webdriverio-docker` and run `npm test`  and see the configuration at https://github.com/serbanghita/webdriverio-docker/blob/master/wdio.conf.js#L58 , you can see that I've delegated `no-sandbox` there. My choice was not to include this in the container, but to find a workaround (user profile, security profile, etc) and don't run the container privileged or mess with the sandbox.

Overall the wrapper solution you provided is good for now but I would like to provide a better solution in the future.

Reply:
The provided solution makes both Chrome and Firefox work (headless or not) without users to have to know or configure anything specific.
Also it's the approach currently used by Selenium official Docker images.